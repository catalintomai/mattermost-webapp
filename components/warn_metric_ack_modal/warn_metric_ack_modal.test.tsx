// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Modal} from 'react-bootstrap';
import {UserProfile} from 'mattermost-redux/src/types/users';

import WarnMetricAckModal from 'components/warn_metric_ack_modal/warn_metric_ack_modal';

jest.mock('react-dom', () => ({
    findDOMNode: () => ({
        blur: jest.fn(),
    }),
}));

describe('components/WarnMetricAckModal', () => {
    const gettingTrialError = 'some error';

    const baseProps = {
        stats: {
            registered_users: 200,
        },
        user: {
            id: 'someUserId',
            first_name: 'Fake',
            last_name: 'Person',
            email: 'a@test.com',
        } as UserProfile,
        show: false,
        diagnosticId: 'diag_0',
        closeParentComponent: jest.fn(),
        warnMetricStatus: {
            id: 'metric1',
            aae_id: 'aae_id',
            limit: 500,
        },
        actions: {
            closeModal: jest.fn(),
            getStandardAnalytics: jest.fn(),
            requestTrialLicenseAndAckWarnMetric: jest.fn(),
            getLicenseConfig: jest.fn()
        },
    };

    test('should match snapshot, init', () => {
        const wrapper = shallow<WarnMetricAckModal>(
            <WarnMetricAckModal {...baseProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('error display', () => {
        const wrapper = shallow<WarnMetricAckModal>(
            <WarnMetricAckModal {...baseProps}/>,
        );

        wrapper.setState({gettingTrialError});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match state when onHide is called', () => {
        const wrapper = shallow<WarnMetricAckModal>(
            <WarnMetricAckModal {...baseProps}/>,
        );

        wrapper.setState({gettingTrial: true});
        wrapper.instance().onHide();
        expect(wrapper.state('gettingTrial')).toEqual(false);
    });

    test('should match state when onHideWithParent is called', () => {
        const wrapper = shallow<WarnMetricAckModal>(
            <WarnMetricAckModal {...baseProps}/>,
        );

        wrapper.setState({gettingTrial: true});
        wrapper.instance().onHideWithParent();

        expect(baseProps.closeParentComponent).toHaveBeenCalledTimes(1);
        expect(wrapper.state('gettingTrial')).toEqual(false);
    });

    test('send ack on acknowledge button click', () => {
        const wrapper = shallow<WarnMetricAckModal>(
            <WarnMetricAckModal {...baseProps}/>,
        );

        wrapper.setState({gettingTrial: true});
        wrapper.find('button').simulate('click');
        expect(baseProps.actions.requestTrialLicenseAndAckWarnMetric).toHaveBeenCalledTimes(1);
    });

    test('should have called props.onHide when Modal.onExited is called', () => {
        const props = {...baseProps};
        const wrapper = shallow(
            <WarnMetricAckModal {...props}/>,
        );

        wrapper.find(Modal).props().onExited!(document.createElement('div'));
        expect(baseProps.actions.closeModal).toHaveBeenCalledTimes(1);
    });
});