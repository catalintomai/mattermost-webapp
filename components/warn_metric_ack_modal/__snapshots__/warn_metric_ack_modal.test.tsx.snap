// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WarnMetricAckModal error display 1`] = `
<Modal
  animation={true}
  aria-labelledby="warnMetricAckHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="warnMetricAckHeaderModalLabel"
    >
      <FormattedMessage
        defaultMessage="Warning"
        id="warn_metric_ack_modal.header.title"
        values={Object {}}
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      <FormattedMessage
        defaultMessage="\\"Team exceeds {limit} users. Consider activating user management controls to ensure compliance. {link}\\""
        id="warn_metric_ack_modal.number_of_active_users.description"
        values={
          Object {
            "limit": 500,
            "link": <ErrorLink
              defaultMessage="aae_id"
              messageId="aae_id"
              url="https://mattermost.com/pl/default-aae-faq"
            />,
          }
        }
      />
      <br />
      <div
        className="form-group has-error"
      >
        <br />
        <label
          className="control-label"
        >
          <FormattedMessage
            defaultMessage="We were unable to reach support. {link} instead to contact support"
            id="warn_metric_ack_modal.mailto.message"
            values={
              Object {
                "link": <WarnMetricAckErrorLink
                  defaultMessage="Email us"
                  messageId="warn_metric_ack_modal.mailto.link"
                  onClickHandler={[Function]}
                  url="mailto:support@mattermost.com?cc=a@test.com&subject=Mattermost%20support%20request%3Aaae_id&body=Mattermost%20support%20request%20for%20aae_id%3A%20Team%20exceeds%20500%20users.%20Consider%20activating%20user%20management%20access%20controls%20to%20ensure%20compliance.%0D%0AContact%20Fake%20Person%0D%0AEmail%20a%40test.com%0D%0ASite%20URL%20http%3A%2F%2Flocalhost%3A8065%0D%0ADiagnostic%20Id%20diag_0%0D%0AIf%20you%20have%20any%20additional%20inquiries%2C%20please%20contact%20support%40mattermost.com"
                />,
              }
            }
          />
        </label>
      </div>
      <br />
      <div
        className="help__format-text"
        style={
          Object {
            "opacity": "0.45",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Contacting support sends your contact information to Mattermost, Inc. {link}"
          id="warn_metric_ack_modal.mailto.body.sub_text"
          values={
            Object {
              "link": <ErrorLink
                defaultMessage="Learn More"
                messageId="warn_metric_ack_modal.learn_more.link"
                url="https://mattermost.com/pl/default-aae-faq"
              />,
            }
          }
        />
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-link cancel-button"
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Cancel"
        id="warn_metric_ack_modal.cancel"
        values={Object {}}
      />
    </button>
    <button
      autoFocus={true}
      className="btn btn-primary save-button"
      data-dismiss="modal"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Contact Support"
        id="warn_metric_ack_modal.contact_support"
        values={Object {}}
      />
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/WarnMetricAckModal should match snapshot, init 1`] = `
<Modal
  animation={true}
  aria-labelledby="warnMetricAckHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="warnMetricAckHeaderModalLabel"
    >
      <FormattedMessage
        defaultMessage="Warning"
        id="warn_metric_ack_modal.header.title"
        values={Object {}}
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      <FormattedMessage
        defaultMessage="\\"Team exceeds {limit} users. Consider activating user management controls to ensure compliance. {link}\\""
        id="warn_metric_ack_modal.number_of_active_users.description"
        values={
          Object {
            "limit": 500,
            "link": <ErrorLink
              defaultMessage="aae_id"
              messageId="aae_id"
              url="https://mattermost.com/pl/default-aae-faq"
            />,
          }
        }
      />
      <br />
      <br />
      <div
        className="help__format-text"
        style={
          Object {
            "opacity": "0.45",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Contacting support sends your contact information to Mattermost, Inc. {link}"
          id="warn_metric_ack_modal.mailto.body.sub_text"
          values={
            Object {
              "link": <ErrorLink
                defaultMessage="Learn More"
                messageId="warn_metric_ack_modal.learn_more.link"
                url="https://mattermost.com/pl/default-aae-faq"
              />,
            }
          }
        />
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-link cancel-button"
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Cancel"
        id="warn_metric_ack_modal.cancel"
        values={Object {}}
      />
    </button>
    <button
      autoFocus={true}
      className="btn btn-primary save-button"
      data-dismiss="modal"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Contact Support"
        id="warn_metric_ack_modal.contact_support"
        values={Object {}}
      />
    </button>
  </ModalFooter>
</Modal>
`;
